File: stone_lib/src/main/java/com/github/klee0kai/stone/annotations/component/GcScopeAnnotation.java
package com.github.klee0kai.stone.annotations.component;


import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation announcing new scopes for garbage collection and changing the caching method.
 * Announce new scopes in the following way.
 *
 * <pre>{@code
 *    ㅤ@GcScopeAnnotation
 *    ㅤ@Retention(RUNTIME)
 *    ㅤ@Target(METHOD)
 *     public @interface GcPlanetScope {
 *     }
 * }</pre>
 * Instead of this annotation, you can use the more general {@link javax.inject.Scope}
 * annotation.
 *
 * <pre>{@code
 *    ㅤ@Scope
 *    ㅤ@Retention(RUNTIME)
 *   ㅤ@Target(METHOD)
 *     public @interface GcPlanetScope {
 *     }
 * }</pre>
 * <p>
 * Not Follow {@link javax.inject.Scope} documentation.
 */

@Retention(value = RetentionPolicy.CLASS)
@Target(value = ElementType.ANNOTATION_TYPE)
public @interface GcScopeAnnotation {
}

File: stone_multiplatform/src/commonMain/kotlin/stone/annotations/component/GcScopeAnnotation.kt
package stone.annotations.component

/**
 * Annotation announcing new scopes for garbage collection and changing the caching method.
 * Announce new scopes in the following way.
 *
 * <pre>`ㅤ@GcScopeAnnotation
 * ㅤ@Retention(RUNTIME)
 * ㅤ@Target(METHOD)
 * public @interface GcPlanetScope {
 * }
`</pre> *
 * Instead of this annotation, you can use the more general [javax.inject.Scope]
 * annotation.
 *
 * <pre>`ㅤ@Scope
 * ㅤ@Retention(RUNTIME)
 * ㅤ@Target(METHOD)
 * public @interface GcPlanetScope {
 * }
`</pre> *
 *
 *
 * Not Follow [javax.inject.Scope] documentation.
 */
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.ANNOTATION_CLASS)
annotation class GcScopeAnnotation 
